pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -68,6 +68,17 @@ func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
 }
 
 func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry, raw json.RawMessage, apiVersion string, resourceID string) api.ValidationResult {
+	log.Infof("running preflight validation on resource: %s", resourceID)
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	isCreate := cosmosdb.IsErrorStatusCode(err, http.StatusNotFound)
+	if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusNotFound) {
+		return api.ValidationResult{
+			Status: api.ValidationStatusFailed,
+			Error: &api.CloudErrorBody{
+				Message: err.Error(),
```

Comment:
If we're special-casing cluster documents not found here, and these errors get sent back to customers, is it better to return a well-defined string, something like "Custer not found for resourceID %s" instead of what I assume will be something related to CosmosDB?